### DISCUSSION TITLE
Installation instructions for Debian

### DISCUSSION THREAD

**Discussion by HenrysCat** [civitai/civitai#570] [Q&A] [answered]
I am trying to install following the instructions on this page https://github.com/civitai/civitai but find them very vague, is there better documentation anywhere?
I get stuck at point 4, where is the .env file? and point 5 gives the error - npm ERR! Missing script: "db:migrate"

**Comment by PeterNjeim** [comment]
The instructions aren't vague, it says:
> Create your .env by making a copy of the contents from .env-example file.

Did you make a copy of `.env-example`?
You can't continue with the steps without setting up your `.env` file, do that first then everything else should work.

**Reply by HenrysCat** [reply]
Where exactly do I find the .env-example file? the file path is what I am looking for here.

**Reply by PeterNjeim** [reply]
I know this sounds insulting, but if you can't find it I'm not sure why you want to build this program? Are you a software developer? The file is in the root directory of the git repo

**Reply by HenrysCat** [reply]
No I am not a software developer, I am just interested in things like this, I know it's in the git repo but when I install on my own server I cannot find it, I know knowledge is power but maybe you could share some of your knowledge.

**Reply by PeterNjeim** [reply]
I am not a gatekeeper of knowledge, I asked if you were a software developer, and also shared where the file was, it is in the root directory of the git repository:

![image](https://github.com/civitai/civitai/assets/26418974/919c73e1-09fc-4f65-a1fd-1b455ef1d72b)

**Reply by HenrysCat** [reply]
I have obviously done something wrong, because in my own install on my own server that file is not there, what could I have done wrong? everything worked up until point 4

**Reply by PeterNjeim** [reply]
You see all of the files except for `.env-example`? Or are you missing even more files and folders

**Reply by HenrysCat** [reply]
I'm probably way out here, I'm looking in /root/.nvm
Am I even close?

**Reply by PeterNjeim** [reply]
Lol not at all, it seems you don't know what "root directory" means either lol. Cloning repositories is meant for software developers, I'm really not sure what you want to do with this repo without knowing how to code, I'm curious. You need to go to the git repository you cloned, that was step 1. It should be a folder called `civitai`, but then again the instructions don't tell you how to clone a repository since that's basic knowledge of software development, so maybe you didn't even clone the repository? I like to help people but I think this is a lost cause rn

**Reply by HenrysCat** [reply]
I assumed as the code was made public I could have a self hosted civitai.com, was I wrong?
Not interested in traffic or 1000's of hits, just a site I can upload my own creations to display.

**Reply by PeterNjeim** [reply, marked as answer]
You can indeed self-host civitai, but the code is public for others to contribute to it, not for self-hosting. Even if you follow all the steps there are some problems with the `.env` file that require you to use an older commit to resolve. If you want to share your works with others, upload them to civitai.com, and share your profile link with others

**Reply by HenrysCat** [reply]
OK thanks
